php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -2095,16 +2110,18 @@ abstract class NDB_BVL_Instrument extends NDB_Page
      */
     function getCandidateAge(string $date): array
     {
-        $dates = array(
-                  'Date_taken' => $this->getFieldValue('Date_taken'),
-                  'DoB'        => $this->getDoB(),
-                 );
+        $dates     = array(
+                      'Date_taken' => $this->getFieldValue('Date_taken'),
+                      'DoB'        => $this->getDoB(),
+                      'DoD'        => $this->getDoD(),
+                     );
+        $dateTaken = empty($date) ? $dates['Date_taken'] : $date;
 
         $age = array();
-        if (empty($date)) {
-            $age = Utility::calculateAge($dates['DoB'], $dates['Date_taken']);
+        if (empty($dates['DoD']) || $dates['DoD'] > $dateTaken) {
```

Comment:
This date comparison should probably be done with `DateTime` objects. Depending on the format of the date string you may get weird results.

Reply:
This is tricky because `calculateAge` expects strings and not `DateTime` objects